sklearn/utils/testing.py

```diff
@@ -622,11 +622,11 @@ def if_matplotlib(func):
     @wraps(func)
     def run_test(*args, **kwargs):
         try:
-            import matplotlib
+            import matplotlib,pyplot as plt
```

Comment:
It should be first ``import matplotlib``, then set the backend, then ``import matplotlib.pyplot``